gems/aws-sdk-core/lib/aws-sdk-core/plugins/telemetry.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Aws
+  module Plugins
+    # @api private
+    class Telemetry < Seahorse::Client::Plugin
+      option(
+        :telemetry_provider,
+        default: Aws::Telemetry::NoOpTelemetryProvider,
+        doc_type: Aws::Telemetry::TelemetryProviderBase,
+        rbs_type: Aws::Telemetry::TelemetryProviderBase,
+        docstring: <<~DOCS) do |_cfg|
+Allows you to provide a telemetry provider, which is used
+to emit telemetry data. By default, uses `NoOpTelemetryProvider`
+which will not record or emit any telemetry data. The SDK currently supports
+OpenTelemetry (OTel) as a telemetry provider. To use the OTel provider, 
+require the`opentelemetry-sdk` gem and then, pass in an instance of a 
+`Aws::Telemetry::OTelProvider` for telemetry provider.
+        DOCS
+        Aws::Telemetry::NoOpTelemetryProvider.new
+      end
+
+      def after_initialize(client)
+        validate_telemetry_provider(client.config)
```

Comment:
If it's set to nil, I think just make it a no-op here.

Reply:
See [here](https://github.com/aws/aws-sdk-ruby/pull/3092/files#r1725679161).